src/frontend/src/tests/lib/canisters/signer.canister.spec.ts

```diff
@@ -0,0 +1,137 @@
+import type { _SERVICE as SignerService, SignRequest } from '$declarations/signer/signer.did';
+import { SignerCanister, type SignerCanisterOptions } from '$lib/canisters/signer.canister';
+import { type ActorSubclass } from '@dfinity/agent';
+import { Principal } from '@dfinity/principal';
+import { mock } from 'vitest-mock-extended';
+import { mockIdentity } from '../../mocks/identity.mock';
+
+vi.mock(import('$lib/constants/app.constants'), async (importOriginal) => {
+	const actual = await importOriginal();
+	return {
+		...actual,
+		LOCAL: false
+	};
+});
+
+describe('signer.canister', () => {
+	const createSignerCanister = async (
+		services: Pick<SignerCanisterOptions<SignerService>, 'serviceOverride'>
+	): Promise<SignerCanister> =>
+		SignerCanister.create({
+			canisterId: Principal.fromText('tdxud-2yaaa-aaaad-aadiq-cai'),
+			identity: mockIdentity,
+			...services
+		});
+	const service = mock<ActorSubclass<SignerService>>();
+
+	beforeEach(() => {
+		vi.clearAllMocks();
+	});
+
+	it('returns correct BTC balance', async () => {
```

Comment:
should we also add some assertions about bubbling the errors in case the service throw an error?

Reply:
Good idea, added them.